src/service/gtfs-flex-service.ts

```diff
@@ -50,35 +36,51 @@ class GtfsFlexService implements IGtfsFlexService {
     }
 
     async getGtfsFlexById(id: string): Promise<FileEntity> {
-        // get a gtfsFlex repository to perform operations with gtfsFlex
-        const gtfsFlexRepository = AppDataSource.getRepository(FlexVersions);
+        const query = {
+            text: 'Select file_upload_path from flex_versions WHERE tdei_record_id = $1',
+            values: [id],
+        }
 
-        // load a gtfsFlex by a given gtfsFlex id
-        const gtfsFlex: FlexVersions | any = await gtfsFlexRepository.findOneBy(
-            {
-                tdei_record_id: id
-            });
+        let result = await dbClient.query(query);
+
+        if (result.rows.length == 0) throw new HttpException(400, "Record not found");
 
         const storageClient = Core.getStorageClient();
         if (storageClient == null) throw console.error("Storage not configured");
-        let url: string = decodeURIComponent(gtfsFlex?.file_upload_path);
+        let url: string = decodeURIComponent(result.rows[0].file_upload_path);
         return storageClient.getFileFromUrl(url);
     }
 
     async createAGtfsFlex(flexInfo: FlexVersions): Promise<GtfsFlexDTO> {
         try {
-            // get a gtfsFlex repository to perform operations with gtfsFlex
-            const gtfsFlexRepository = AppDataSource.getRepository(FlexVersions);
             flexInfo.file_upload_path = decodeURIComponent(flexInfo.file_upload_path!);
-            // create a real gtfsFlex object from gtfsFlex json object sent over http
-            const newGtfsFlex = gtfsFlexRepository.create(flexInfo);
+            const queryObject = {
```

Comment:
Please write the escape characters for the string type of values. This will otherwise lead to possible SQL injection

Reply:
basically in service layer , we pass this input as parameterised and it takes care of escaping the characters